sydent/threepid/bind.py

```diff
@@ -101,29 +102,27 @@ def _notify(self, assoc, attempt):
         mxid = assoc["mxid"]
         domain = mxid.split(":")[-1]
         server = yield self._pickServer(domain)
-        callbackUrl = "https://%s/_matrix/federation/v1/3pid/onbind" % (
+
+        post_url = "https://%s/_matrix/federation/v1/3pid/onbind" % (
             server,
         )
 
         logger.info("Making bind callback to: %s", callbackUrl)
-        # TODO: Not be woefully insecure
-        agent = Agent(reactor, InsecureInterceptableContextFactory())
-        reqDeferred = agent.request(
-            "POST",
-            callbackUrl.encode("utf8"),
-            Headers({
+
+        # Make a POST to the chosen Synapse server
+        http_client = FederationHttpClient()
+        response = yield http_client.post_json_get_nothing(post_url, assoc, {
```

Comment:
[the docs on Agent](https://twistedmatrix.com/documents/current/web/howto/client.html) (which also apply to FederationAgent, since they share an interface) have this to say:

> An important thing to keep in mind is that the body will only be read from the connection after Response.deliverBody is called. This also means that the connection will remain open until this is done (and the body read). So, in general, any response with a body must have that body read using deliverBody . If the application is not interested in the body, it should issue a HEAD request or use a protocol which immediately calls stopProducing on its transport.

In other words, we're going to leak HTTP connections because we fail to read the response. 

If you look at things like `post_json_get_json` in the synapse you'll notice that it is careful to always call `readBody` (which in turn calls `deliverBody`). I'd suggest you cargo-cult that function into the sydent codebase.